internal/mode/static/handler.go

```diff
@@ -194,6 +173,11 @@ func (h *eventHandlerImpl) HandleEventBatch(ctx context.Context, logger logr.Log
 	case state.EndpointsOnlyChange:
 		h.version++
 		cfg := dataplane.BuildConfiguration(ctx, gr, h.cfg.serviceResolver, h.version)
+		depCtx, setErr := h.setDeploymentCtx(ctx, logger)
+		if setErr != nil {
+			logger.Error(setErr, "error setting deployment context for usage reporting")
```

Comment:
missing unit tests for this line. Not sure if codecov is showing old results

Reply:
We typically don't test log output.